internal/assertion/assertion.go

```diff
@@ -49,19 +49,20 @@ func (assertion *Assertion) NotTo(matcher types.GomegaMatcher, optionalDescripti
 }
 
 func (assertion *Assertion) buildDescription(optionalDescription ...interface{}) string {
-	switch len(optionalDescription) {
-	case 0:
+	if len(optionalDescription) == 0 {
 		return ""
-	default:
-		return fmt.Sprintf(optionalDescription[0].(string), optionalDescription[1:]...) + "\n"
 	}
+	if describe, ok := optionalDescription[0].(func() string); ok && len(optionalDescription) == 1 {
```

Comment:
It might benchmark as faster to do the original `switch len(optionalDescription)`, with this new `if` condition in the `case 1:` block.
That way you check the length once, and only attempt the type assertion `.(func() string)` when length is known to be 1.

Reply:
thanks for reviewing @jim-slattery-rs. It makes sense what you suggest, we pushed another commit with the change :)